ccmlib/node.py

```diff
@@ -836,30 +879,37 @@ def start(self,
             self.__clean_win_pid()
             self._update_pid(process)
             print_("Started: {0} with pid: {1}".format(self.name, self.pid), file=sys.stderr, flush=True)
-        elif update_pid:
-            self._update_pid(process)
 
-            if not self.is_running():
-                raise NodeError("Error starting node %s" % self.name, process)
+        if update_pid or wait_for_binary_proto:
+            # at this moment we should have PID and it should be running...
+            if not self._wait_for_running(process, timeout_s=7):
+                raise NodeError("Node {n} is not running".format(n=self.name), process)
 
-        # If wait_other_notice is a bool, we don't want to treat it as a
-        # timeout. Other intlike types, though, we want to use.
-        if common.is_intlike(wait_other_notice) and not isinstance(wait_other_notice, bool):
+        # if requested wait for other nodes to observe this one (via gossip)
+        if common.is_int_not_bool(wait_other_notice):
             for node, mark in marks:
                 node.watch_log_for_alive(self, from_mark=mark, timeout=wait_other_notice)
         elif wait_other_notice:
             for node, mark in marks:
                 node.watch_log_for_alive(self, from_mark=mark)
 
-        # If wait_for_binary_proto is a bool, we don't want to treat it as a
-        # timeout. Other intlike types, though, we want to use.
-        if common.is_intlike(wait_for_binary_proto) and not isinstance(wait_for_binary_proto, bool):
+        # if requested wait for binary protocol to start
+        if common.is_int_not_bool(wait_for_binary_proto):
```

Comment:
what about composing two variables 

```python
should_wait_for_binary_proto = wait_for_binary_proto is not None
binary_proto_timeout = wait_for_binary_proto if is_int_not_bool(wait_for_binary_proto) else None
``` 

and then just call 

```python
if should_wait_for_binary_proto: 
    self.wait_for_binary_interface(from_mark=self.mark, timeout=binary_proto_timeout)
```

Reply:
this will not work straight because `wait_for_binary_proto` can be `False`.

Reply:
I would rather not touch it...